packages/teams-js/src/public/meeting.ts

```diff
@@ -560,4 +591,78 @@ export namespace meeting {
       sendMessageToParent('meeting.appShareButton.setOptions', [shareInformation]);
     }
   }
+
+  /**
+   * Ext App will handle the Audio(Mic & Speaker). Teams to go Audio less (No Audio)
+   *
+   * @throws standard parameter validation error
+   * @param requestAppAudioHandlingParams - RequestAppAudioHandlingParams interface, and contains
+   *   isAppHandlingAudio - true - expect App to handle the audio and Teams to go audioless.
+   *     false - Audio transferred to Teams and Teams to restore the audio.
+   *   callbackMicMuteStateChangedHandler - Callback contains mic status param.
+   * @param callback - Callback contains IRequestAppAudioHandlingSdkResponse param, error or operation result.
+   * error can either contain an error of type SdkError (error indication), or null (non-error indication)
+   * result can either contain a true boolean value (successful termination), or null (unsuccessful fetch)
+   * @beta
+   */
+  export function requestAppAudioHandling(
+    requestAppAudioHandlingParams: RequestAppAudioHandlingParams,
+    callback: (
+      error: SdkError | null,
+      requestAppAudioHandlingSdkResponse: IRequestAppAudioHandlingSdkResponse | null,
+    ) => void,
+  ): void {
+    if (!callback) {
+      throw new Error('[requestAppAudioHandling] Callback response cannot be null');
+    }
+    if (!requestAppAudioHandlingParams.callbackMicMuteStateChangedHandler) {
+      throw new Error('[requestAppAudioHandling] Callback Mic mute state handler cannot be null');
+    }
+    ensureInitialized(runtime, FrameContexts.sidePanel, FrameContexts.meetingStage);
+
+    const callbackInternalRequest = (
+      error: SdkError | null,
+      requestAppAudioHandlingSdkResponse: IRequestAppAudioHandlingSdkResponse | null,
+    ): void => {
+      if (!error) {
+        registerHandler('meeting.micStateChanged', requestAppAudioHandlingParams.callbackMicMuteStateChangedHandler);
+      }
+      callback(error, requestAppAudioHandlingSdkResponse);
+    };
+
+    const callbackInternalStop = (
+      error: SdkError | null,
+      requestAppAudioHandlingSdkResponse: IRequestAppAudioHandlingSdkResponse | null,
+    ): void => {
+      if (doesHandlerExist('meeting.micStateChanged')) {
+        removeHandler('meeting.micStateChanged');
+      }
+      callback(error, requestAppAudioHandlingSdkResponse);
+    };
+
+    if (requestAppAudioHandlingParams.isAppHandlingAudio) {
+      sendMessageToParent(
+        'meeting.requestAppAudioHandling',
+        [requestAppAudioHandlingParams.isAppHandlingAudio],
+        callbackInternalRequest,
+      );
+    } else {
+      sendMessageToParent(
+        'meeting.requestAppAudioHandling',
+        [requestAppAudioHandlingParams.isAppHandlingAudio],
+        callbackInternalStop,
+      );
+    }
+  }
+
+  /**
+   * Send Mic mute status response acknowledgement for callbackMicMuteStateChangedHandler
+   *
+   * @param micState - Mic mute ack
```

Comment:
"Mic mute ack" is not a descriptive enough comment for public facing documentation. Please use @link to reference where a developer should use this

Reply:
Also, why do you have the user pass this callback into requestAppAudioHandling? Are there cases where an app developer would want to specify a custom handler for this? If not maybe requestAppAudioHandling should just register this callback internally so that app developers don't have to know about it